### PR TITLE
[WIP] Add eslint

### PR DESCRIPTION
**Pull request by koddsson**
Get our codebase linted and nice.

Note that I'm rebasing this branch against mercilessly since it's gonna be somewhat longliving. So if you plan on checking it out just ping me and I can stop rebasing it and just make sure to squash all the commits before merging.

TODO:
- [x] Add lint command to CI
- [x] Lint all the things
- [ ] Replace `lib/cache.js` with [express-redis-cache](https://www.npmjs.com/package/express-redis-cache)?

### CODE REVIEW

**Comment by kristjanmik** [comment]
Super good job! Looking forward to this one!

**Comment by koddsson** [comment]
I someone wants to work on this with me, feel free! Just make sure that you let me know what part you're working on so we don't do the work twice. :dancers:

**Comment by koddsson** [comment]
I just disabled the linting on travis for this branch so I can catch any test failures when they come up more easily.

**Comment by benediktvaldez** [comment]
FYI: Working on some major updates to this branch

File: endpoints/carparks/index.js [hunk 1]
```diff
@@ -1,53 +1,57 @@
...
+      })
+    }
 
     try {
-      var data = $(body)
-    } catch (error) {
-      return res.status(500).json({ error:'Could not parse body' })
+      // FIXME: What is the point of this? It doesn't seem to be used anywhere
+      // eslint-disable-next-line no-unused-vars
+      const data = $(body)
```
Comment (benediktvaldez) [inline endpoints/carparks/index.js]: @haraldurkarls I stumbled upon this while updating this file to be lint-compliant

What is the point of this? It doesn't seem to be used anywhere

**Comment by benediktvaldez** [comment]
And we're down to `✖ 39 problems (0 errors, 39 warnings)`

**Comment by benediktvaldez** [comment]
And we're down to `✖ 39 problems (0 errors, 39 warnings)`

**Comment by benediktvaldez** [comment]
BAMM! Clean lint output! Fixed linting for all files (see (endless) commit history for details) - codeship and travis are now both running `npm run lint` before testing..



This was only made possible due to the fact I have other things to do that are quite more important and time sensitive, but my procrastinating nature pushed me this way.. (sorry jsconf)

I suggest merging this, and moving the replacement of lib/cache to it's own branch/PR - what do you say @koddsson?

**Comment by kristjanmik** [comment]
You guys are my heroes @benediktvaldez and @koddsson !

**Comment by koddsson** [comment]
>I suggest merging this, and moving the replacement of lib/cache to it's own branch/PR - what do you say @koddsson?

Definitely!

**Comment by koddsson** [comment]
I just merged this with the worst commit message :disappointed:

**Pull request merged**